better list layout and bug fixes

shivanshPurple opened the pull request:
- Implemented the native Command Palette loading indicator (`IsLoading`)
- Updated the message handling to call `UpdateMessagesMemo()` and `RaiseItemsChanged()` during streaming so responses appear instantly. 

These changes are kinda vibe coded. Please let me know if any changes or other things required.

shivanshPurple commented:
@LioQing requesting you to review this PR. And merge so I can officially enjoy the loading bar 😄

<img width="798" height="489" alt="image" src="https://github.com/user-attachments/assets/b69e3064-568f-4cc8-9d48-10146d0fbb37" />

LioQing reviewed (commented):
thanks for the PR, i understand the streaming update, but for loading bar i thought it was already there before?

LioQing commented on README.md:
I'm not sure about these steps... at least that's not how I do it

shivanshPurple replied:
Sure, I will remove it then. Its probably because I didn't use visual studio to build this. I was using vscode with command line build commands.

shivanshPurple commented:
> thanks for the PR, i understand the streaming update, but for loading bar i thought it was already there before?

@LioQing You are right. I just checked the available version again and loading is there. I don't know how I hallucinated and believed that it wasn't there before.
Streaming also seems to be there previously. idk how did I not notice.

Please allow me some time to understand the code more. And I will make changes to the PR then or just straight up close the PR.

shivanshPurple committed:
better list layout and bug fixes

shivanshPurple commented:
I spent some time refining the extension to make it feel smoother. I found that real-time streaming caused way too much flickering in the Command Palette UI, so I didn't do it.

The changes:

- Didn't do the real-time streaming. The list now just says "Generating..." while loading, and the full answer pops in when ready.
- Swapped the list items so your Question is the main title (easier to scan history) and the subtitle is a short preview. (I felt we shouldn't show the answers twice in the same small window)
- Added a guard to prevent double-submitting if you spam the Enter key.
- Fixed an issue where older messages that got stuck/failed would look like duplicate "Press enter to send" buttons. They now correctly show "No response received".

---

Please review when you have a chance. Apologies for any oversights, as I'm still getting used to the open-source workflow.
Also, let me know if these changes aren't needed. I’m happy to close the PR and maintain them on my fork.

shivanshPurple committed:
Optimize chat history building with Take() for cleaner code

LioQing commented:
Sorry for being so unresponsive because of busy things going on in life, but now as I took a good look at your changes, they are pretty solid, I'll merge them 👍

shivanshPurple commented:
Hey, no worries. Thanks a lot!
